clients/src/main/resources/common/message/DescribeTopicPartitionsRequest.json

```diff
@@ -30,6 +30,8 @@
     },
     { "name": "ResponsePartitionLimit", "type": "int32", "versions": "0+", "default": "2000",
       "about": "The maximum number of partitions included in the response." },
+    { "name": "IncludeTopicAuthorizedOperations", "type": "bool", "versions": "1+",
```

Comment:
I think you're along the right lines, but I think there's a problem with this approach. KIP-966 introduced the DescribeTopicPartitions RPC and it does indeed not include `IncludeTopicAuthorizedOperations`. But that KIP introduced v0 of the RPC, and it would take another KIP to introduce v1. It's quite hard to figure out which KIP introduced which RPC versions, so I might have missed it.

I suggest a code-only change in the admin client which only fills in the authorized operations given to the client if the client did request them. Of course, the RPC response contains them regardless of whether they were requested.

Reply:
Thanks Andrew for pointing it out. You are right, this can be solved by code-only change in the admin client change.